source/merge.ts

```diff
@@ -15,9 +15,9 @@ export default function merge(target: Options, ...sources: Options[]): Options {
 				target[key] = new URL(sourceValue as string, targetValue);
 			} else if (is.plainObject(sourceValue)) {
 				if (is.plainObject(targetValue)) {
-					target[key] = merge({}, targetValue as Options, sourceValue as Options);
+					target[key] = merge({}, targetValue as { [key: string]: unknown }, sourceValue as { [key: string]: unknown });
```

Comment:
This type assertion is needed because `is.plainObject` returns a `boolean` and can't be used as a type guard to check if it is an object or an object literal.

Reply:
@fnky Should we change https://github.com/sindresorhus/is/blob/ab586df0f9ff91e4c0e0d692a978cba110df1d7c/source/index.ts#L224 to 

```ts
export const plainObject = (value: unknown) => value is {[key: string]: unknown} {
```
?

Reply:
Yeah I think that should work :)

Reply:
https://github.com/sindresorhus/is/pull/80